bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java

```diff
@@ -477,8 +477,8 @@ private BArray getRecordArray(TomlNode tomlValue, String variableName, ArrayType
             recordName = type.getName();
             recordType = (RecordType) type;
         } else {
-            recordName = ((BIntersectionType) type).getEffectiveType().getName();
-            recordType = (RecordType) ((BIntersectionType) type).getEffectiveType();
+            recordType = (RecordType) ((BIntersectionType) type).getConstituentTypes().get(0);
```

Comment:
Shall we create an issue to revisit the assumption that this will always be the first member?

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/30178